### PR TITLE
TBC Redis caching implementation

### PR DESCRIPTION
**Pull request by timolegros**
## Link to Issue
Closes: #5749

## Description of Changes
- Implements/updates the `getKeys` and `setKeys` methods of the Redis controller
- Adds caching for EVM/Cosmos balances
- Adds a `cacheRefresh` option to `getBalances` used to forcibly refresh the cache
- Added test cases for caching balances

## Test Plan
- None

## Deployment Plan

1. 

## Other Considerations

Should be quickly followed up with #5804 and #5807

### CODE REVIEW

**Comment by timolegros** [comment]
> Can you add a test plan?

Unfortunately in this standalone PR there isn't a way to manually test the TBC caching. Only the new CI tests cover the caching functionality. Alternatively, you can merge this PR into your open PR and run the refresh script but that won't guarantee the returned balances are correct unless you manually check them of course.

File: packages/commonwealth/server/util/tokenBalanceCache/tokenBalanceCache.ts [hunk 1]
```diff
@@ -118,31 +142,120 @@ export class TokenBalanceCache {
...
+          addresses: validatedAddresses,
           contractAddress: options.sourceOptions.contractAddress,
           tokenId: options.sourceOptions.tokenId,
         });
+        break;
     }
+
+    await this.cacheBalances(options, freshBalances);
+
+    return { ...freshBalances, ...cachedBalances };
```
Comment (rbennettcw) [inline packages/commonwealth/server/util/tokenBalanceCache/tokenBalanceCache.ts:186]: Shouldn't the fresh balances be spread *after* the cached balances?

**Review by rbennettcw — APPROVED** [review]
LGTM to test with refresh

File: packages/commonwealth/server/util/tokenBalanceCache/tokenBalanceCache.ts (hunk 1 above)
Comment (rbennettcw) [reply packages/commonwealth/server/util/tokenBalanceCache/tokenBalanceCache.ts:186]: It's fine if there's no overlap in keys between the fresh and cached objects. If there is overlap, then the cached values will override the fresh.

**Review by ianrowan — APPROVED** [review]
Look good

**Pull request merged**